week7/answers.sql

```diff
@@ -1,3 +1,314 @@
+	character_id INT UNSIGNED,
+	item_id INT UNSIGNED,
+  
+	CONSTRAINT inventory_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE,
+  
+	CONSTRAINT inventory_fk_items
+		FOREIGN KEY (item_id)
+		REFERENCES items (item_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+CREATE TABLE equipped (
+	equipped_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	character_id INT UNSIGNED,
+	item_id INT UNSIGNED,
+  
+	CONSTRAINT equipped_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE,
+  
+	CONSTRAINT equipped_fk_items
+		FOREIGN KEY (item_id)
+		REFERENCES items (item_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+DELIMITER ;;
+CREATE PROCEDURE equip(inventory_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT inventory_id INTO find_item_by_id;
+
+	SELECT 
+		inventory.character_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		inventory.item_id
+	FROM inventory
+	WHERE inventory.inventory_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO equipped (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM inventory WHERE inventory.inventory_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE unequip(equipped_id INT UNSIGNED)
+BEGIN
+	DECLARE find_item_by_id INT UNSIGNED;
+	DECLARE id_of_character INT UNSIGNED;
+	DECLARE id_of_item INT UNSIGNED;
+
+	SELECT equipped_id INTO find_item_by_id;
+
+	SELECT 
+		equipped.character_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_character;
+
+	SELECT 
+		equipped.item_id
+	FROM equipped
+	WHERE equipped.equipped_id = find_item_by_id
+	INTO id_of_item;
+
+	INSERT INTO inventory (character_id, item_id) VALUES (id_of_character, id_of_item);
+
+	DELETE FROM equipped WHERE equipped.equipped_id = find_item_by_id; 
+END;;
+
+CREATE PROCEDURE set_winners(team_id INT UNSIGNED)
+BEGIN
+	DECLARE find_winner_by_team_id INT UNSIGNED;
+	SELECT team_id INTO find_winner_by_team_id;
+
+	DELETE FROM winners;
+
+	INSERT INTO winners
+	SELECT 
+		characters.character_id,
+		characters.name
+	FROM teams
+		INNER JOIN team_members
+			ON teams.team_id = team_members.team_id
+		INNER JOIN characters
+			ON team_members.character_id = characters.character_id
+	WHERE teams.team_id = find_winner_by_team_id;
+END;;
+
+CREATE FUNCTION armor_total(character_id INT UNSIGNED)
+RETURNS TINYINT
+DETERMINISTIC 
+BEGIN
+	DECLARE find_armor_total_by_character_id INT UNSIGNED;
+	DECLARE armor_from_items TINYINT;
+	DECLARE natural_armor TINYINT;
+
+	SELECT character_id INTO find_armor_total_by_character_id;
+
+	SELECT 
+		SUM(items.armor)
+	FROM equipped
+		INNER JOIN items
+			ON equipped.item_id = items.item_id
+	WHERE equipped.character_id = find_armor_total_by_character_id
```

Comment:
Why not just:

```sql
WHERE equipped.character_id = char_id
```

(Assuming you rename your input variable to `char_id` (or nearly any name besides `character_id`).) This will save the need for the `find_armor_total_by_character_id` variable and the explicit copy on line 192.